### PR TITLE
LPD-40575  | Import attachments from URL

### PR DESCRIPTION
**Pull request by jaimelr10**
**Description**:
As suggested in: https://github.com/liferay-headless/liferay-portal/pull/2291 

> Problem:
>
>Currently, Liferay object entries require attachments to be uploaded as base64 or link it with an existing Document in the document library. This can be cumbersome, especially when dealing with large files or when attachments are stored remotely.
>
>Solution:
>
>Introduce a new property to the attachment field type schema: attachmentURL. (we can discuss on the name)
>
>When importing an object entry, if the attachmentURL property is filled, Liferay will:
>
>Fetch the document from the specified URL.
>
>Store the downloaded document in the document library.
>
>Associate the downloaded document with the object entry.
>
>The new mechanism supports both batch imports and individual object entry creation through the POST endpoint.

**Jira Ticket**: https://liferay.atlassian.net/browse/LPD-40575

### CODE REVIEW

**Comment by jaimelr10** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
@jaimelr10 The failing tests seem related to the attachment functionality. Can you review?

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java [hunk 1]
```diff
@@ -1378,14 +1387,39 @@ private void _processAttachment(
...
+				if (Objects.equals(url.getProtocol(), "file")) {
+					throw new UnsupportedOperationException(
+						"Unsupported protocol");
+				}
+
+				URLConnection urlConnection = url.openConnection();
+
+				urlConnection.connect();
+
+				fileContent = StreamUtil.toByteArray(
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java:1410]: I think you should set the field to null once it has been downloaded as we do with the base64:

```java
fileEntry.setFileBase64(() -> (String)null);
```

**Review by ccorreagg — COMMENTED** [review]
Please check my comments. The tests seem to be failing

**Comment by jaimelr10** [comment]
ci:test:relevant

**Comment by jaimelr10** [comment]
ci:test:relevant

**Comment by jaimelr10** [comment]
ci:test:relevant

**Comment by jaimelr10** [comment]
ci:test:relevant

**Comment by jaimelr10** [comment]
ci:test:relevant

**Comment by jaimelr10** [comment]
ci:test:relevant

**Comment by jaimelr10** [comment]
ci:test:relevant

**Comment by jaimelr10** [comment]
ci:test:relevant

**Comment by jaimelr10** [comment]
ci:test:relevant

**Review by 4lejandrito — APPROVED** [review]
@jaimelr10 read my changes carefully.

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
ci:forward:force

**Comment by 4lejandrito** [comment]
ci:forward:force

**Pull request closed without merging**